clippy_lints/src/doc.rs

```diff
@@ -609,3 +694,43 @@ fn check_word(cx: &LateContext<'_>, word: &str, span: Span) {
         );
     }
 }
+
+struct FindPanicUnwrap<'a, 'tcx> {
+    lcx: &'a LateContext<'tcx>,
+    typeck_results: &'tcx ty::TypeckResults<'tcx>,
```

Comment:
Since you pass the `LateContext` anyway, this field shouldn't be necessary. Calling `cx.tcx.expr_ty(..)` is also more stable.

Reply:
The problem is that we get the typecheck results form `cx.tcx.typeck(impl_item_def_id)` and otherwise we would have to have a def_id field